examples/md-flexible/CMakeLists.txt

```diff
@@ -5,30 +5,32 @@ file(
         "src/*.h"
 )
 
-set(AUTOPAS_INCLUDE_MPI "$ENV{AUTOPAS_INCLUDE_MPI}")
+include_directories(SYSTEM ${MPI_INCLUDE_PATH})
+
+add_executable(md-flexible ${MDFlex_SRC})
+
+target_include_directories(md-flexible PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
+include(autopas_yaml-cpp)
+
+option(AUTOPAS_INCLUDE_MPI "Links MPI libraries to the target. This is required if you want to use MPI." OFF)
 
 if (AUTOPAS_INTERNODE_TUNING OR AUTOPAS_INCLUDE_MPI)
     message(STATUS "AUTOPAS_INTERNODE_TUNING set, searching for mpi to link against md-flexible ...")
     find_package(MPI)
 
     if (NOT ${MPI_CXX_FOUND})
         message(FATAL_ERROR "cxx mpi could not be found, even though AUTOPAS_INTERNODE_TUNING was set.")
+        target_link_libraries(md-flexible PUBLIC autopas autopasTools yaml-cpp)
     else ()
         message(STATUS "cxx mpi found: ${MPI_CXX_COMPILER}")
+        target_link_libraries(md-flexible PUBLIC autopas autopasTools yaml-cpp ${MPI_CXX_LIBRARIES})
     endif ()
 else ()
+    target_link_libraries(md-flexible PUBLIC autopas autopasTools yaml-cpp)
     message(STATUS "AUTOPAS_INTERNODE_TUNING not set, not linking MPI to md-flexible.")
```

Comment:
```suggestion

add_executable(md-flexible ${MDFlex_SRC})

target_include_directories(md-flexible PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})

include(autopas_yaml-cpp)

option(AUTOPAS_INCLUDE_MPI "Links MPI libraries to the target. This is required if you want to use MPI." OFF)

if (AUTOPAS_INTERNODE_TUNING OR AUTOPAS_INCLUDE_MPI)
    message(STATUS "AUTOPAS_INTERNODE_TUNING set, searching for mpi to link against md-flexible ...")
    find_package(MPI)
    include_directories(SYSTEM ${MPI_INCLUDE_PATH})

    if (NOT ${MPI_CXX_FOUND})
        message(FATAL_ERROR "cxx mpi could not be found, even though AUTOPAS_INTERNODE_TUNING was set.")
    else ()
        message(STATUS "cxx mpi found: ${MPI_CXX_COMPILER}")
        set(MD-FLEXIBLE_USE_MPI ON)
    endif ()
else ()
    message(STATUS "AUTOPAS_INTERNODE_TUNING not set, not linking MPI to md-flexible.")
endif ()

# MPI Variables undefined if find_package fails
target_link_libraries(md-flexible PUBLIC autopas autopasTools yaml-cpp $<$<BOOL:${MD-FLEXIBLE_USE_MPI}>:MPI::MPI_CXX>)

```

Reply:
might conflict with the next suggestion but github does not let me include that line into this suggestion.
Also: Test this thoroughly before trusting me ;)